File: packages/jexlang-java/src/main/java/com/jexlang/java/eval/errors/JexLangSyntaxError.java
package com.jexlang.java.eval.errors;

public class JexLangSyntaxError extends JexLangRuntimeError {

    private final SyntaxErrorLocation location;

    public JexLangSyntaxError(String message, SyntaxErrorLocation location) {
        super(message);
        this.location = location;
    }

    public SyntaxErrorLocation getLocation() {
        return location;
    }

    @Override
    public String getName() {
        return "JexLangSyntaxError";
    }
}

File: packages/jexlang-swift/jexlang-swift/eval/errors/JexLangSyntaxError.swift
//
//  JexLangSyntaxError.swift
//  jexlang-swift
//
//  Created by Vinay Kumar on 03/10/25.
//

import Foundation

public class JexLangSyntaxError: JexLangRuntimeError {
    private let location: SyntaxErrorLocation
    
    public init(message: String, location: SyntaxErrorLocation) {
        self.location = location
        super.init(message: message)
    }
    
    public override func getName() -> String {
        return "JexLangSyntaxError"
    }
}
